crates/sdk/src/local.rs

```diff
@@ -0,0 +1,144 @@
+use crate::mode::Mode;
+use crate::opts::ProofOpts;
+use crate::proof::SP1ProofWithPublicValues;
+use crate::prover::Prover;
+use crate::provers::SP1VerificationError;
+use crate::request::{ProofRequest, DEFAULT_TIMEOUT};
+use crate::verify;
+
+use anyhow::Result;
+use async_trait::async_trait;
+use futures::executor;
+use sp1_core_executor::{ExecutionError, ExecutionReport, SP1Context};
+use sp1_core_machine::io::SP1Stdin;
+use sp1_primitives::io::SP1PublicValues;
+use sp1_prover::components::DefaultProverComponents;
+use sp1_prover::{SP1Prover, SP1ProvingKey, SP1VerifyingKey, SP1_CIRCUIT_VERSION};
+use std::future::{Future, IntoFuture};
+use std::pin::Pin;
+
+pub struct LocalProver {
+    pub prover: SP1Prover<DefaultProverComponents>,
+}
+
+impl LocalProver {
+    pub fn new() -> Self {
+        Self { prover: SP1Prover::new() }
+    }
+
+    pub fn builder() -> LocalProverBuilder {
+        LocalProverBuilder::new()
+    }
+}
+
+pub struct LocalProverBuilder {}
+
+impl LocalProverBuilder {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    pub fn build(self) -> LocalProver {
+        LocalProver::new()
+    }
+}
+
+pub struct LocalProofRequest<'a> {
+    pub prover: &'a LocalProver,
+    pub pk: &'a SP1ProvingKey,
+    pub stdin: SP1Stdin,
+    pub mode: Mode,
+    pub timeout: u64,
+}
+
+impl<'a> LocalProofRequest<'a> {
+    pub fn new(prover: &'a LocalProver, pk: &'a SP1ProvingKey, stdin: SP1Stdin) -> Self {
+        Self { prover, pk, stdin, timeout: DEFAULT_TIMEOUT, mode: Mode::default() }
+    }
+
+    pub fn with_mode(mut self, mode: Mode) -> Self {
+        self.mode = mode;
+        self
+    }
+
+    pub fn with_timeout(mut self, timeout: u64) -> Self {
+        self.timeout = timeout;
+        self
+    }
+
+    pub async fn run(self) -> Result<SP1ProofWithPublicValues> {
```

Comment:
`run` methods should not be async

Reply:
but they need to `await` in their logic

Reply:
For the networkprover right? If the underlying client exposes blocking functions use that, otherwise youll need. to spawn a new exeucutor and call block on

LocalProver shouldnt need async to run, the point of run is to be the blocking (not async) entrypoint